YoutubeTest/app/src/main/java/com/nulp/youtubetest/DataList.java

```diff
@@ -0,0 +1,87 @@
+package com.nulp.youtubetest;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.widget.LinearLayout;
+
+import com.google.android.material.snackbar.Snackbar;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class DataList extends AppCompatActivity {
+
+    private RecyclerView recyclerView;
+    private SwipeRefreshLayout swipeRefreshLayout;
+    private LinearLayout linearLayout;
+    private EquipmentAdapter adapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_data_list);
+
+        initViews();
+        registerNetworkMonitoring();
+        loadEquipment();
+
+    }
+
+    private void registerNetworkMonitoring() {
+        IntentFilter filter = new IntentFilter("android.net.conn.CONNECTIVITY_CHANGE");
+        NetworkChangeReceiver receiver = new NetworkChangeReceiver(linearLayout);
+        this.registerReceiver(receiver, filter);
+    }
+
+    private void initViews() {
+        recyclerView = findViewById(R.id.data_list_recycler_view);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        linearLayout = findViewById(R.id.linearLayout);
+        swipeRefreshLayout = findViewById(R.id.data_list_swipe_refresh);
+        setupSwipeToRefresh();
+    }
+
+    private void loadEquipment() {
+        swipeRefreshLayout.setRefreshing(true);
+        final ApiService apiService = getApplicationEx().getApiService();
+        final Call<List<Equipment>> call = apiService.getEquipment();
+
+        call.enqueue(new Callback<List<Equipment>>() {
+            @Override
+            public void onResponse(final Call<List<Equipment>> call,
+                                   final Response<List<Equipment>> response) {
+                adapter = new EquipmentAdapter(response.body());
```

Comment:
No need to create adapter every time you load new data. It has to be initialized just once